[BUG] 배포환경에서 ElasticSearch가 연결되지 않는 오류 수정

## 🐞 버그 설명
ElasticSearch에 인가된 CA 인증서 검증 로직을 생략하는 코드가 localhost에만 적용되어 있어, 배포 환경에서 ElasticSearch와 관련된 Bean이 생성되지 않는 문제를 해결합니다.

## 📋 재현 방법
버그를 재현하는 단계:
- 배포환경에서 실행하면 Spring이 실행되지 않습니다.

## ⚙️ 기대 동작
- 배포환경에서 Spring이 정상적으로 실행되어야합니다.

## 🖼️ 스크린샷
스크린샷 대신에 관련 로그를 첨부 합니다.
```
Caused by: org.springframework.dao.DataAccessResourceFailureException: Connection refused
	at org.springframework.data.elasticsearch.client.elc.ElasticsearchExceptionTranslator.translateExceptionIfPossible(ElasticsearchExceptionTranslator.java:111) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.client.elc.ElasticsearchExceptionTranslator.translateException(ElasticsearchExceptionTranslator.java:65) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.client.elc.ChildTemplate.execute(ChildTemplate.java:73) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.doExists(IndicesTemplate.java:177) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.exists(IndicesTemplate.java:169) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.createIndexAndMappingIfNeeded(SimpleElasticsearchRepository.java:93) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.repository.support.SimpleElasticsearchRepository.<init>(SimpleElasticsearchRepository.java:87) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:499) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:480) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:208) ~[spring-beans-6.1.10.jar:6.1.10]
	... 76 common frames omitted
Caused by: java.lang.RuntimeException: Connection refused
	at org.springframework.data.elasticsearch.client.elc.ElasticsearchExceptionTranslator.translateException(ElasticsearchExceptionTranslator.java:64) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	... 87 common frames omitted
Caused by: java.net.ConnectException: Connection refused
	at org.elasticsearch.client.RestClient.extractAndWrapCause(RestClient.java:934) ~[elasticsearch-rest-client-8.13.4.jar:8.13.4]
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:304) ~[elasticsearch-rest-client-8.13.4.jar:8.13.4]
	at org.elasticsearch.client.RestClient.performRequest(RestClient.java:292) ~[elasticsearch-rest-client-8.13.4.jar:8.13.4]
	at co.elastic.clients.transport.rest_client.RestClientHttpClient.performRequest(RestClientHttpClient.java:91) ~[elasticsearch-java-8.13.4.jar:na]
	at co.elastic.clients.transport.ElasticsearchTransportBase.performRequest(ElasticsearchTransportBase.java:144) ~[elasticsearch-java-8.13.4.jar:na]
	at co.elastic.clients.elasticsearch.indices.ElasticsearchIndicesClient.exists(ElasticsearchIndicesClient.java:666) ~[elasticsearch-java-8.13.4.jar:na]
	at org.springframework.data.elasticsearch.client.elc.IndicesTemplate.lambda$doExists$2(IndicesTemplate.java:177) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	at org.springframework.data.elasticsearch.client.elc.ChildTemplate.execute(ChildTemplate.java:71) ~[spring-data-elasticsearch-5.3.1.jar:5.3.1]
	... 86 common frames omitted
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method) ~[na:na]
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672) ~[na:na]
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946) ~[na:na]
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvent(DefaultConnectingIOReactor.java:174) ~[httpcore-nio-4.4.16.jar:4.4.16]
	at org.apache.http.impl.nio.reactor.DefaultConnectingIOReactor.processEvents(DefaultConnectingIOReactor.java:148) ~[httpcore-nio-4.4.16.jar:4.4.16]
	at org.apache.http.impl.nio.reactor.AbstractMultiworkerIOReactor.execute(AbstractMultiworkerIOReactor.java:351) ~[httpcore-nio-4.4.16.jar:4.4.16]
	at org.apache.http.impl.nio.conn.PoolingNHttpClientConnectionManager.execute(PoolingNHttpClientConnectionManager.java:221) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at org.apache.http.impl.nio.client.CloseableHttpAsyncClientBase$1.run(CloseableHttpAsyncClientBase.java:64) ~[httpasyncclient-4.1.5.jar:4.1.5]
	at java.base/java.lang.Thread.run(Thread.java:833) ~[na:na]
```

## ✅ TODO
- [ ] 배포환경에서 ElasticSearch가 연결되지 않는 오류 수정

--- a/src/main/java/com/leets/commitatobe/global/config/elastic/ElasticSearchConfig.java
+++ b/src/main/java/com/leets/commitatobe/global/config/elastic/ElasticSearchConfig.java
@@ -26,6 +26,9 @@ public class ElasticSearchConfig {
 	@Value("${spring.elasticsearch.password}")
 	private String password;
 
+	@Value("${hostname}")
+	private String hostname;
+
 	@Bean
 	public ElasticsearchClient elasticSearchClient() {
 		SSLContext sslContext;
@@ -41,7 +44,7 @@ public class ElasticSearchConfig {
 		credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(username, password));
 
 		RestClient restClient = RestClient.builder(
-				new HttpHost("localhost", 9200, "https"))
+				new HttpHost(hostname, 9200, "https"))
 			.setHttpClientConfigCallback(httpClientBuilder ->
 				httpClientBuilder.setSSLContext(sslContext)
 					.setSSLHostnameVerifier(NoopHostnameVerifier.INSTANCE)

--- a/src/main/resources/application-dev.yaml
+++ b/src/main/resources/application-dev.yaml
@@ -59,3 +59,5 @@ management:
       application: ${spring.application.name}
 
 server-uri: ENC(9uSGkzaRV41gMgfy74TVxkbrDYtD0CuIDFSRJuQba14hudMDsWXVhw==)
+
+hostname: ENC(5H01n5YFZSpX7RchqwfV1yMJ4aRIhyXg)

--- a/src/main/resources/application-local.yaml
+++ b/src/main/resources/application-local.yaml
@@ -58,3 +58,5 @@ management:
       application: ${spring.application.name}
 
 server-uri: http://localhost:8080
+
+hostname: localhost

--- a/src/main/resources/application-prod.yaml
+++ b/src/main/resources/application-prod.yaml
@@ -59,3 +59,5 @@ management:
       application: ${spring.application.name}
 
 server-uri: ENC(ugcPjCCqPIKcaBDy9haOfZilhoZ/l2WM1vpO5WjxC2CSIgYPy6s/KQ==)
+
+hostname: ENC(8xM3quuM8SRP+x7PJ0QQdIHcJP6LaPXZ)

